### PR TITLE
configure: fail if PSL is not disabled but not found

### LINKED ISSUE #14373: configure no longer errors when libpsl is not found and not explicitly disabled
### I did this

```
./configure --without-ssl
```

### I expected the following

```
configure: error: libpsl was not found
```

### curl/libcurl version

8.9.1

### operating system

macOS 12

### PR DESCRIPTION
**Pull request by bagder**
Regression since 9b3f67e (shipped in 8.7.0)
Reported-by: Ryan Carsten Schmidt
Fixes #14373

### CODE REVIEW

**Comment by bagder** [comment]
It is quite hard to find the configure line in these failing builds. We should probably do something to make it easier. Most failures seem to be legitimate: they don't find libpsl.

**Comment by bagder** [comment]
@vszakats I'm getting lost in these workflow files. Can you help me figure out where we need to add libpsl to make the configure runs work?

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -449,7 +449,7 @@ jobs:
       - name: 'install autotools'
         if: ${{ matrix.build == 'autotools' }}
         run: |
-          echo autoconf automake libtool openssl | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
+          echo autoconf automake libtool openssl libpsl | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
```
Comment (vszakats) [inline .github/workflows/macos.yml:452]:
```suggestion
          echo autoconf automake libtool openssl pkg-config libpsl | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
```
Perhaps this, based on how it's done in the autotools and cmake jobs.

Comment (vszakats) [reply .github/workflows/macos.yml:452]: Nope, works without `pkg-config`:
https://github.com/curl/curl/actions/runs/10237324011/job/28320483945

Comment (vszakats) [inline .github/workflows/macos.yml:452]:
```suggestion
          echo autoconf automake libtool openssl | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
```

It works with autotools now, but cmake doesn't pick it up and it's intentional.
The 'combinations' jobs are meant to be minimal and match between autotools and cmake. So I think the correct solution is to use `--without-libpsl` here.

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -586,6 +586,7 @@ jobs:
               --disable-dependency-tracking \
               --disable-docs --disable-manual \
               --without-nghttp2 --without-libidn2 \
+              --with-libpsl=$(brew --prefix libpsl) \
```
Comment (vszakats) [inline .github/workflows/macos.yml:589]:
```suggestion
              --without-libpsl \
```

**Pull request closed without merging**